### PR TITLE
Feature/142 assign roles review

### PR DESCRIPTION
**Pull request by AidanLaycock**

### CODE REVIEW

**Comment by tobytwigger** [comment]
@AidanLaycock This is amazing, it's such an improvement! Literally my only comment is that, when adding a new role, the role title should change if the position changes. Possibly in a watcher (there may be a better way), if the text in role title matches the old position, then set it to the newly selected position

File: public/modules/assign-roles/js/module.js [hunk 1]
```diff
@@ -2127,10 +2127,9 @@ __webpack_require__.r(__webpack_exports__);
     };
   },
   watch: {
-    positionId: function positionId(val) {
-      if (this.role_name === null) {
-        this.role_name = this.getPositionName(val);
-      }
+    positionId: function positionId(newVal, oldVal) {
+      // If role name is empty or hasn't been changed then assign Position Name
+      return this.role_name === null || this.role_name === this.getPositionName(oldVal) ? this.role_name = this.getPositionName(newVal) : '';
```
Comment (tobytwigger) [inline public/modules/assign-roles/js/module.js]: Can we also check if the role name is ''? If you click in the role name box before changing the position, it doesn't change as `this.role_name` = '' not null

File: resources/js/components/participant/AddRoleForm.vue [hunk 2]
```diff
@@ -66,11 +66,10 @@
         },
 
         watch: {
-            positionId(val) {
-                if(this.role_name === null) {
-                    this.role_name = this.getPositionName(val);
-                }
-            }
+            positionId(newVal, oldVal) {
+                // If role name is empty or hasn't been changed then assign Position Name
```
Comment (tobytwigger) [inline resources/js/components/participant/AddRoleForm.vue:70]: Same here.

**Review by tobytwigger — CHANGES_REQUESTED** [review]
Minor minor bug I spotted, other than that this is ready for merge!

**Comment by AidanLaycock** [comment]
@tobytwigger - Both of the above issues should be resolved. I've refactored it to incorporate Null/falsey Values.

**Review by tobytwigger — APPROVED** [review]

**Pull request merged**